Feat/dev 159 detect secrets

asurinsaka opened the pull request:
DEV-159 add pre-commit and detect-secrets

We would like to use pre-commit to perform various checks for us. 
Detecting secrets before committing is one use-case.

philtom-ctds reviewed (changes requested):
Need to fix and update master before merging this pr.  I've opened prs to address master.  Blocking this pr for now

asurinsaka commented:
> Need to fix and update master before merging this pr. I've opened prs to address master. Blocking this pr for now

I do not need to do anything for this one, do I?

asurinsaka committed:
DEV-159 add pre-commit and detect-secrets

Add pre-commit and detect-secrets so it starts to automatically detect
secrets for this code base.
pre-commit requires cfgv. Use old cfgv version to work with py35 and py27

asurinsaka committed:
DEV-159 add baseline

Add baseline file for detect-secrets for future reference.

asurinsaka committed:
DEV-159 update readme

Update table of contents.

asurinsaka committed:
Update travis.yaml to use python 3.5

change python 3.4 to python 3.5
